test/reporters/base.spec.js

```diff
@@ -403,4 +404,16 @@ describe('Base reporter', function() {
     errOut = stdout.join('\n').trim();
     expect(errOut, 'to be', '1) test title:\n     Error\n  foo\n  bar');
   });
+
+  it('should let you stub out console.log without effecting reporters output', function() {
+    sinon.stub(console, 'log');
```

Comment:
1. a sinon sandbox is typically easier to use, because no matter what you do, it's only one `restore()` call:

	```js
	var sandbox;
	var myStub;
	var myOtherStub;
	
	beforeEach(function() {
	  sandbox = sinon.createSandbox();
	  myStub = sandbox.stub();
	  myOtherStub = sandbox.stub();
	})
	
	afterEach(function() {
	  sandbox.restore();
	});
	```

2. if you call `restore()` in the test itself after making assertions, the call won't happen if the assertions fail; use hooks instead

Reply:
```js
  describe('when reporter output immune to user test changes', function() {
    var sandbox;

    beforeEach(function() {
      sandbox = sinon.createSandbox();
      sinon.stub(console, 'log');
      sinon.stub(Base, 'consoleLog');
    });

    it('should let you stub out console.log without effecting reporters output', function() {
      Base.list([]);

      expect(Base.consoleLog, 'was called');
      expect(console.log, 'was not called');
    });

    afterEach(function() {
      sandbox.restore();
    });
  });
```

Reply:
Nice with the sandboxes. But if I don't restore the `Base.consoleLog` before the test ends it breaks the reporter.

Reply:
Why?

Reply:
I guess as its interfering with running which is running the test.

> node bin/mocha "test/reporters/base.spec.js"

Stub inside hooks, restore inside test
```
  Base reporter
    when reporter output immune to user test changes
      ✓ should let you stub out console.log without effecting reporters output

  1 passing (17ms)
```

Stub + restore inside hooks
```
  Base reporter
    when reporter output immune to user test changes

  1 passing (17ms)
```

Thoughts?

Reply:
Edited sandbox code above.